.github/workflows/ci.yml

```diff
@@ -134,10 +134,18 @@ jobs:
         shell: bash
         run: echo ::set-output name=flavor::$([[ "${{ matrix.os }}" == "cuda" ]] && echo "Cuda" || echo "CPU")
 
+      - name: Set environment variables
```

Comment:
```suggestion
      # Clean tests (kernel caching disabled) use more CPU and memory, so we only run them for
      # scheduled runs, pushes to master, and pushes to tags.
      # Parallelization also needs to be disabled to avoid Out of Memory errors.
      - name: Set environment variables
```

Reply:
Thank you both @MoFtZ and @jgiannuzzi for these comments, much appreciated! 🍀 
Since the initial idea happened over a call, my bad was that started immediately working on PR for Ivan's repo instead of writing down those requirements.
I'll update this PR once the change is confirmed and tested by our team.

Reply:
here comes the latest changes, as you suggested 🚀